pallets/gear/src/lib.rs

```diff
@@ -809,21 +814,24 @@ pub mod pallet {
                             context,
                             pages_with_data,
                         } => {
-                            let memory_pages = if lazy_pages_enabled {
+                            #[cfg(feature = "lazy-pages")]
+                            let memory_pages = {
+                                let _ = pages_with_data;
+                                assert!(lazy_pages::try_to_enable_lazy_pages());
```

Comment:
We had talks about using here `try_to_enable...` every block and it's really incorrect. We should hide this functionality on the node cli maybe

Reply:
`try_to_enable...` must be in `on_idle`

Reply:
And also we must to avoid to call it if block has no contracts to execute.

Reply:
> `try_to_enable...` must be in `on_idle`

why should we do it everytime?

Reply:
Because `on_idle` can be called in separate threads. And we must set signal handler for each thread separately. In future for multithread execution we will must to do it for each `on_idle`, not only for each thread.